File: sdks/php/src/Resources/NotificationsResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Notification;
use Puxbay\Models\PaginatedResponse;

class NotificationsResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("notifications/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => Notification::fromArray($item));
    }

    public function getNotification(string $notificationId): Notification
    {
        $data = $this->get("notifications/{$notificationId}/");
        return Notification::fromArray($data);
    }

    public function markAsRead(string $notificationId): Notification
    {
        $data = $this->post("notifications/{$notificationId}/mark-read/");
        return Notification::fromArray($data);
    }
}

File: sdks/dotnet/src/Resources/NotificationsResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class NotificationsResource
    {
        private readonly PuxbayClient _client;
        public NotificationsResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Notification>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<Notification>>($"notifications/?page={page}");
        }

        public async Task<Notification> GetAsync(string notificationId)
        {
            return await _client.GetAsync<Notification>($"notifications/{notificationId}/");
        }

        public async Task<Notification> MarkAsReadAsync(string notificationId)
        {
            return await _client.PostAsync<Notification>($"notifications/{notificationId}/mark-read/", null);
        }
    }
}
